### ISSUE TITLE
Fire Emblem 3 houses crashes on the latest update: some shaders are causing the game to crash

### ISSUE THREAD

**Issue by Destroyer4LP** [Ryujinx/Ryujinx#1621]
### What's the issue you encountered?

I updated Ryujinx.
On the latest update (Ryujinx Version: 1.0.5525) the game Fire Emblem 3 houses crashes if you use Crest analyzer. You could interact with the Crest analyzer on versions before without crashing.
Later I discovered it happens at other event too. I thing something broke some shaders. The Game crashes 100% of the time it try to fix it.

I added my savegame for you to test it faster.

### How can the issue be reproduced?

In the game Fire Emblem 3 houses if  you use Crest analyzer the game crashes.
If you enter Black Magic Tournament and use "Fire" the same thing happens (probally also in battles).
In the church you can do the chor activity with also crashes the game.

If you do any of these things the screen freezes and you can hear the sound for 0.5s-2s befor crashing.

### Log file + savestate

[Ryujinx_2020-10-14_21-30-50.log](https://github.com/Ryujinx/Ryujinx/files/5386420/Ryujinx_2020-10-14_21-30-50.log)
[slot08.zip](https://github.com/Ryujinx/Ryujinx/files/5386528/slot08.zip)

### Environment?

- Ryujinx version: 1.0.5525.X</br>

- Game version: 1.2.0</br>
Fire Emblem 3 houses
- System Specs:
  - OS: Windows 10
  - CPU: i7-5820k
  - GPU: NVIDIA GTX 1070
  - RAM: 16GB
- Applied Mods : [ FireEmblemThreeHouses-60FPS-1.2.0 ]

### Additional context?

Additional info about your environment:</br>

**Comment by gdkchan** [comment]
Potentially fixed by #1618.

**Comment by Destroyer4LP** [comment]
I builded ryujinx manuelly and tested if game still crashes. It crashed again.
Maby did something wrong building ryijinx ( 515 Warnings, 0 Errors ).

**Comment by gdkchan** [comment]
Log?

**Comment by Destroyer4LP** [comment]
Took me awhile. This game Takes long to load on ryujinx.
[Ryujinx_2020-10-15_20-11-43.log](https://github.com/Ryujinx/Ryujinx/files/5386959/Ryujinx_2020-10-15_20-11-43.log)

**Comment by AcK77** [comment]
First you should disable OpenGL Debug Logs since it’s slow down the execution and it’s not needed for you.
Second, the log file didn’t contains any error message, but thanks for the issue!

**Comment by caloyers** [comment]
Its still crashing in version 1.0.5545 when

1.  You do the cathedral choir Song
2.  In the scene where you attack  your opponent(garbage sound with slow jittery video) 
3. Even when leaving it on the Opening screen without doing anything wherein the opening movie plays and showing characters classes, the after a minute or two Ryujinx will close.
4. Sometimes when loading screen in battle, it will crash.

Using Ryujinx Version 1.0.5485 issue 1 does not crash.

**Comment by Destroyer4LP** [comment]
@AcK77 
> First you should disable OpenGL Debug Logs since it’s slow down the execution and it’s not needed for you.

I know I only turned it on because I was asked for my log. So the thing with the load times comments from me is, I have seen three times the Emulator took about 3 min before I saw the loading screen of the game.(performance optimized... logs off) I even accidentally restarted the emulator couple of time because I thought Ryujinx was somehow frozen. I don't know if somehow windows did something weird or if there was an issue with ryujinx.
Currently I did not experience such behaviour. There were some updates in that time frame.

> Second, the log file didn’t contains any error message, but thanks for the issue!

In the log you can see as soon as you trigger the bug and the Game frezzes:
"00:03:23.734 |W| GUI.RenderLoop Gpu Log: Shader translator: Invalid instruction at 0x0029B0 (0xED01000002101E1F)."
The stuff after this is write in the log ms before the emulator crashes.

Idk why the bug doesn't cause an Error to be written in the log. Maybe the info can help you.

**Comment by crazyen** [comment]
I have experienced the same issues in version 1.0.5525 and 1.0.5545. It happened in various situations and I used the cathedral choir Song for quick crash testing.  Then I use Version 1.0.5496 which has no such issues. I remember I updated one or two times after 1.0.5496 before theses problems are encountered but I don't remember the last working version because I used the auto update function and did not download from the website.

However, there are still other minor issues in all versions. I have listed them in the compatibility list page. 
[https://github.com/Ryujinx/Ryujinx-Games-List/issues/230](url)

Mostly the game runs near perfectly. Audio works fine, images are sharp. Sometimes a little bit lagging in the quick action animation in the battlefield though.

**Comment by gdkchan** [comment]
The PR that fixes some of those issues was merged, please retest and update the issue (or close if fixed) if possible, thanks.

**Comment by Destroyer4LP** [comment]
@gdkchan 
The Magic Tournament work now fine. At first, the first Battle took about 3 Minutes (normally about 10-15s) and the Game did only show about 40 Frames. But after the first battle, it runs smoothly with about 50fps. Much better than before.
The Crest analyzer now works fine. I could not test the cathedral choir Song but I think it should work fine.
Thank you!

**Comment by crazyen** [comment]
The cathedral choir Song still causes Ryujinx to crash in version 1.0.5573.

> The PR that fixes some of those issues was merged, please retest and update the issue (or close if fixed) if possible, thanks.

In a new game plus, the first game battle is OK. However, after loading a random save and it still crashes on any attack in the battle while Version 1.0.5496 works fine.

**Comment by Destroyer4LP** [comment]
I retested everything because I thought I did miss something. 
But my game works still fine:

- weekly battles work fine

- tournaments work

- cathedral choir Song works

- Cest analyzer works

I reopended this Issue but can't replicate your issues.
Ryujinx Version: 1.0.5573

**Comment by crazyen** [comment]
I retested the cathedral choir song using Version 1.0.5573 three times and today it worked fine. There was a lagging in the first test.

Then I loaded the battle save failed yesterday, and it crashed. This time I loaded another battle save and everthing worked just fine.  After reopening Ryujinx, I loaded the failed battle save again and it didn't crash this time. It is possible that the crashes are relevant to the cache system because some lagging could be noticed in the first successful battle and choir song animations.  I am sorry that I didn't test thoroughly yesterday.

Finally I tested the issue mentioned in Ryujinx/Ryujinx-Games-List#230 and it is now fixed. Dorothea (dancer class) can now use the dance ability. Thank the developers for the great work.
